### ISSUE TITLE
How to replicate the bottom-up-top-down results in the original paper?

### ISSUE THREAD

**Issue by wzn0828** [ruotianluo/self-critical.pytorch#48]
How to set hyperparameters to replicate the results in the paper-"Bottom-Up and Top-Down Attention for Image Captioning"?

**Comment by ruotianluo** [comment]
just set the network size as in the paper, leave else the default, you can get close result.

**Comment by wzn0828** [comment]
@ruotianluo Thank you very much.

**Comment by wzn0828** [comment]
@ruotianluo Hi, I am sorry I still can't replicate the results with the same network size as in the paper. In fact I just get a 1.060 Cider under XE loss training. Could you give the correct configuration to replicate the results?

**Comment by ruotianluo** [comment]
Do you use beam search to test?

**Comment by wzn0828** [comment]
@ruotianluo 
I didn't use beam search because an AttributeError was reported while using it: 
![image](https://user-images.githubusercontent.com/29533484/42357597-464ad320-810b-11e8-9216-dd958537d007.png)
Will the beam search raise the cider from to my 1.060 to 1.135 reported in the paper ? I think that is a big gap.

The other question, when I set opt.drop_prob_lm to the default 0.5 or 0.6, the model is still overfitting:
![image](https://user-images.githubusercontent.com/29533484/42357695-bf00cf40-810b-11e8-8988-44561f965295.png)
The green is using drop_prob_lm to 0.5, and the other 0.6.

The follows is my configuration:
opt.id = 'topdown'
opt.caption_model = 'topdown'
opt.rnn_size = 1000
opt.input_encoding_size = 1000

opt.batch_size = 16
opt.optim = 'adam'
opt.learning_rate = 5e-4
opt.optim_alpha = 0.9
opt.beam_size = 1
opt.learning_rate_decay_start = 0
opt.drop_prob_lm = 0.5

opt.checkpoint_path = 'Experiments/top-down-original-debug'
opt.save_checkpoint_every = 6000
opt.val_images_use = 5000
opt.max_epochs = 30
opt.language_eval = 1

Could you tell me my question? Many thanks for your help.

**Comment by wzn0828** [comment]
I campare yours to the official code, there are a little difference: you add an affine to the original features before using them in language generator. But I do not think so that is it. Anyway, I will try it. 

I have added the code "os.environ['CUDA_VISIBLE_DEVICES'] = '0'", but it did not work. 

I will try att2in2 model. Thank you for your help.

**Comment by coldmanck** [comment]
@wzn0828 Did you use the bottom-up features to train the model?

**Comment by wzn0828** [comment]
@coldmanck Yes, I used.  But I still couldn't replicate the paper's result with the paper's set such as the rnn size. However I get a close result with a shrinkaged rnn_size.
